### PR TITLE
update gradle to 8

### PR DESCRIPTION
**Pull request by kishansairam9**

### CODE REVIEW

File: gradle/wrapper/gradle-wrapper.properties [hunk 1]
```diff
@@ -1,5 +1,7 @@
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-7.3.2-all.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-8.3-all.zip
```
[the remark this answers is not in the document]
Comment (kishansairam9) [reply gradle/wrapper/gradle-wrapper.properties:3]: 8.4 didn't work on my machine failing to fetch distribution from gradle site. Might be connection issue, but happened multiple times
```sh
HEAD request to https://services.gradle.org/distributions/gradle-8.4-all.zip failed: response code (401)
```

[the remark this answers is not in the document]
Comment (aaron-steinfeld) [reply gradle/wrapper/gradle-wrapper.properties:3]: huh bizarre, 401. Shouldn't be anything version specific but don't worry about it if it's still failing.

**Comment by aaron-steinfeld** [comment]
huh why is build and dependency-check not required? let me fix that.

**Review by aaron-steinfeld — DISMISSED** [review]

File: .trivyignore [hunk 2]
```diff
@@ -0,0 +1,2 @@
+# libssl3
+CVE-2023-5678 exp:2023-12-31
```
Comment (kishansairam9) [inline .trivyignore:2]: image
didn't report under PR tab but guessing this is the cause (based on other repo observations)

**Review by aaron-steinfeld — DISMISSED** [review]

**Review by aaron-steinfeld — DISMISSED** [review]

**Comment by aaron-steinfeld** [comment]
Raw spans grouper is using client utils (the grpc channel registry) without declaring it. It's getting it implicitly from kafka streams framework. Declare the dep and should be good (could also update kafka stream framework, not sure if latest version has the change but we'd want to add the explicit declaration regardless).

**Review by aaron-steinfeld — APPROVED** [review]

**Pull request merged**